uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -78,65 +78,84 @@
 
 use anyhow::{bail, Result};
 
-use super::record::Field;
+use super::record::{Field, FieldDescr};
 use super::types::Type;
-use super::{APIConverter, ComponentInterface};
+use super::{APIConverter, ComponentInterface, CINode};
 
 /// Represents an enum with named variants, each of which may have named
 /// and typed fields.
 ///
 /// Enums are passed across the FFI by serializing to a bytebuffer, with a
 /// i32 indicating the variant followed by the serialization of each field.
+#[derive(Debug)]
+pub struct Enum<'a> {
+    pub(super) parent: &'a ComponentInterface,
+    pub(super) descr: &'a EnumDescr,
+}
+
 #[derive(Debug, Clone, Hash)]
-pub struct Enum {
+pub struct EnumDescr {
     pub(super) name: String,
-    pub(super) variants: Vec<Variant>,
+    pub(super) variants: Vec<VariantDescr>,
     // "Flat" enums do not have, and will never have, variants with associated data.
     pub(super) flat: bool,
 }
 
-impl Enum {
+impl<'a> Enum<'a> {
     pub fn name(&self) -> &str {
-        &self.name
+        &self.descr.name
     }
 
-    pub fn variants(&self) -> Vec<&Variant> {
-        self.variants.iter().collect()
+    pub fn variants(&self) -> Vec<Variant<'_, Self>> {
+        self.descr
+            .variants
+            .iter()
+            .map(|v| Variant {
+                parent: self,
+                descr: v,
+            })
+            .collect()
     }
 
     pub fn is_flat(&self) -> bool {
-        self.flat
+        self.descr.flat
     }
 
-    pub fn contains_object_references(&self, ci: &ComponentInterface) -> bool {
-        // *sigh* at the clone here, the relationship between a ComponentInterace
-        // and its contained types could use a bit of a cleanup.
-        ci.type_contains_object_references(&Type::Enum(self.name.clone()))
+    pub fn contains_object_references(&self) -> bool {
+        self.variants().iter().any(|v| {
+            v.contains_object_references()
+        })
     }
 
-    pub fn contains_unsigned_types(&self, ci: &ComponentInterface) -> bool {
-        self.variants().iter().any(|v| {
-            v.fields()
+    pub fn contains_unsigned_types(&self, _ci: &ComponentInterface) -> bool {
```

Comment:
What's the reason to keep `_ci` here? Is it necessary or just work-in-progress for now?